engine/src/cmd/json.h

```diff
@@ -0,0 +1,850 @@
+#ifndef JSON_H
+#define JSON_H
+
+/*! \file json.h
+ * \brief Simpleson header file
+ */
+
+#include <cstdlib>
+#include <string>
+#include <vector>
+#include <cstdio>
+#include <utility>
+#include <stdexcept>
+#include <cctype>
+
+/*! \brief Base namespace for simpleson */
+namespace json
```

Comment:
Is there a reason we're implementing our own JSON parser?
Can't we just use Boost's JSON Parser (https://www.boost.org/doc/libs/1_79_0/boost/property_tree/json_parser.hpp)?

Reply:
@royfalk are you using this library? https://github.com/gregjesl/simpleson

Reply:
That may make sense...but then we need to document it; however, that Boost::Json_Parser is more actively maintained and used by a larger audience, so less likely to have issues and we wouldn't have to maintain/sync it.

Reply:
I'm fine with using simpleson, personally. The code is already written to use that library. Why bother changing it? Also, I seem to recall reading that Boost's JSON parser leaves something to be desired.

We do need to figure out why the CI is failing, though.

Reply:
@BenjamenMeyer Issue filed as #686

Reply:
@BenjamenMeyer you will need to approve the PR, or I will not be able to merge.

Reply:
@stephengtuggy thanks! @royfalk will approve